lib/libvol2bird.c

```diff
@@ -149,8 +153,74 @@ PolarVolume_t* vol2birdGetIRISVolume(char* filenames[], int nInputFiles);
 
 PolarVolume_t* vol2birdGetODIMVolume(char* filenames[], int nInputFiles);
 
+#ifdef VOL2BIRD_R
+int check_mistnet_loaded_c(void);
+#endif
+
+static vol2bird_printfun vol2bird_internal_printf_fun = vol2bird_default_print;
+
+static vol2bird_printfun vol2bird_internal_err_printf_fun = vol2bird_default_err_print;
+
 // non-public function declarations (local to this file/translation unit)
 
+void vol2bird_printf(const char* fmt, ...)
+{
+  va_list ap;
+  int n;
+  char msg[65536];
+  va_start(ap, fmt);
+  n = vsnprintf(msg, 1024, fmt, ap);
+  va_end(ap);
+  if (n >= 0 && n <= 65536) {
+    vol2bird_internal_printf_fun(msg);
+  } else {
+    vol2bird_internal_printf_fun("vol2bird_printf failed when printing message");
+  }
+}
+
+void vol2bird_err_printf(const char* fmt, ...)
+{
+  va_list ap;
+  int n;
+  char msg[65536];
+  va_start(ap, fmt);
+  n = vsnprintf(msg, 1024, fmt, ap);
+  va_end(ap);
+  if (n >= 0 && n <= 65536) {
+    vol2bird_internal_err_printf_fun(msg);
+  } else {
+    vol2bird_internal_err_printf_fun("vol2bird_err_printf failed when printing message");
+  }
+}
+
+void vol2bird_default_print(const char* msg)
+{
+#ifndef NO_VOL2BIRD_PRINTF
```

Comment:
Do we need to set/define this flag in Makefiles?

Reply:
No, only used when building vol2birdR.